webcompat/webhooks/helpers.py

```diff
@@ -206,3 +230,177 @@ def new_opened_issue(payload):
         headers=headers,
         data=json.dumps(payload_request))
     return proxy_response
+
+
+def process_issue_action(issue_info):
+    """Route the actions and provide different responses.
+
+    There are two possible known scopes:
+    * public repo
+    * private repo
+
+    Currently the actions we are handling are (for now):
+    * opened (public repo only)
+      Aka newly issues created and
+      need to be assigned labels and milestones
+    * milestoned (private repo only)
+      When the issue is being moderated with a milestone: accepted
+    """
+    source_repo = issue_info['repository_url']
+    scope = repo_scope(source_repo)
+    issue_number = issue_info['number']
+    # We do not process further in case
+    # we don't know what we are dealing with
+    if scope == 'unknown':
+        return ('Wrong repository', 403, {'Content-Type': 'text/plain'})
+    if issue_info['action'] == 'opened' and scope == 'public':
+        # we are setting labels on each new open issues
+        response = tag_new_public_issue(issue_info)
+        if response.status_code == 200:
+            return ('gracias, amigo.', 200, {'Content-Type': 'text/plain'})
+        else:
+            msg_log('public:opened labels failed', issue_number)
+            return ('ooops', 400, {'Content-Type': 'text/plain'})
+    # TODO probably do a function. Too many conditions.
+    elif (issue_info['action'] == 'milestoned' and
+          scope == 'private' and
+          issue_info['milestoned_with'] == 'accepted'):
+        # private issue have been moderated and we will make it public
+        response = private_issue_moderation(issue_info)
+        if response.status_code == 200:
+            return ('Moderated issue accepted',
+                    200, {'Content-Type': 'text/plain'})
+        else:
+            msg_log('private:moving to public failed', issue_number)
+            return ('ooops', 400, {'Content-Type': 'text/plain'})
+    elif (scope == 'private' and issue_info['state'] == 'closed'):
+        # private issue has been closed. It is rejected
+        # We need to patch with a template.
+        response = private_issue_rejected(issue_info)
+        if response.status_code == 200:
+            return ('Moderated issue rejected',
+                    200, {'Content-Type': 'text/plain'})
+        else:
+            msg_log('public rejection failed', issue_number)
+            return ('ooops', 400, {'Content-Type': 'text/plain'})
+    else:
+        return ('Not an interesting hook', 403, {'Content-Type': 'text/plain'})
+
+
+def repo_scope(source_repo):
+    """Check the scope nature of the repository.
+
+    The repository can be a
+    * known public.
+    * known private.
+    * or completely unknown.
+    """
+    scope = 'unknown'
+    if source_repo.endswith(PUBLIC_REPO.rsplit('/issues')[0]):
```

Comment:
so i thought about that **but** if at any point github allows a pattern in the uri which matches this `a in b` we will be weaker. So that was a preemptive defensive choice  I made on the code.